### PR TITLE
Book and Category tests implementation

### PR DESCRIPTION
**Pull request by ADIGrimm**

### CODE REVIEW

File: src/test/java/online/bookstore/book/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @MockBean
+    private JwtAuthenticationFilter jwtAuthenticationFilter;
+    @MockBean
+    private JwtUtil jwtUtil;
```
Comment (liliia-ponomarenko) [inline src/test/java/online/bookstore/book/BookControllerTest.java:61]: Why do you need it?

File: src/test/java/online/bookstore/book/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(
+            scripts = "classpath:database/books/delete-new-book.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Sql(
+            scripts = "classpath:database/books/delete-fiction-category.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @Transactional
```
Comment (liliia-ponomarenko) [inline src/test/java/online/bookstore/book/BookControllerTest.java]: if your test doesn’t work without this annotation `@SqlMergeMode(SqlMergeMode.MergeMode.MERGE)` can help you

File: src/test/java/online/bookstore/book/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        requestBook.setCategories(Set.of(category));
+
+        BookDto expected = new BookDto();
+        expected.setTitle(requestBook.getTitle());
+        expected.setAuthor(requestBook.getAuthor());
+        expected.setPrice(requestBook.getPrice());
+        expected.setIsbn(requestBook.getIsbn());
+        expected.setDescription(requestBook.getDescription());
+        expected.setCoverImage(requestBook.getCoverImage());
+        expected.setCategories(requestBook.getCategories());
```
Comment (liliia-ponomarenko) [inline src/test/java/online/bookstore/book/BookControllerTest.java]: better move such logic to the private method or util class and simplify your test cases.

File: src/test/java/online/bookstore/book/BookServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    private BookMapper bookMapper;
+    @Mock
+    private BookSpecificationBuilder bookSpecificationBuilder;
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    public void getAllBooks_WithPageable_ShouldReturnAllBooks() {
+        Pageable pageable = PageRequest.of(0, 10);
+        List<Book> books = List.of(new Book(), new Book());
```
Comment (liliia-ponomarenko) [inline src/test/java/online/bookstore/book/BookServiceTest.java:43]: better initialize books. Also, I recommend you use Given-When-Than pattern

File: src/test/java/online/bookstore/category/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
```
Comment (liliia-ponomarenko) [inline src/test/java/online/bookstore/category/CategoryControllerTest.java:46]: the same comments

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

File: src/test/java/online/bookstore/book/BookControllerTest.java (hunk 1 above)
Comment (ADIGrimm) [reply src/test/java/online/bookstore/book/BookControllerTest.java:61]: Because tests fail without jwt.secret, which i don't know how to fix

**Review by okuzan — APPROVED** [review]

**Pull request merged**